diff --git a/src/hotspot/share/memory/memoryReserver.cpp b/src/hotspot/share/memory/memoryReserver.cpp
index d68736c7f2a..069264de2d3 100644
--- a/src/hotspot/share/memory/memoryReserver.cpp
+++ b/src/hotspot/share/memory/memoryReserver.cpp
@@ -437,7 +437,7 @@ ReservedSpace HeapReserver::Instance::try_reserve_range(char *highest_start,
 
   // Try attach points from top to bottom.
   for (char* attach_point = highest_start;
-       attach_point >= lowest_start && attach_point <= highest_start;  // Avoid wrap around.
+       attach_point >= lowest_start;
        attach_point -= stepsize) {
     ReservedSpace reserved = try_reserve_memory(size, alignment, page_size, attach_point);
 
@@ -450,6 +450,8 @@ ReservedSpace HeapReserver::Instance::try_reserve_range(char *highest_start,
 
       release(reserved);
     }
+    if (p2u(attach_point) <= stepsize)
+      break;  //Make UndefinedBehaviorSanitizer silent
   }
 
   // Failed
