diff --git a/make/hotspot/lib/CompileJvm.gmk b/make/hotspot/lib/CompileJvm.gmk
index 49a96ec3273..d83e4e43cf6 100644
--- a/make/hotspot/lib/CompileJvm.gmk
+++ b/make/hotspot/lib/CompileJvm.gmk
@@ -200,6 +200,9 @@ $(eval $(call SetupJdkLibrary, BUILD_LIBJVM, \
     DISABLED_WARNINGS_gcc_synchronizer.cpp := stringop-overflow, \
     DISABLED_WARNINGS_gcc_templateInterpreterGenerator_x86.cpp := unused-const-variable, \
     DISABLED_WARNINGS_gcc_xGlobals_ppc.cpp := unused-const-variable, \
+    DISABLED_WARNINGS_gcc_instanceKlass.cpp := stringop-overflow, \
+    DISABLED_WARNINGS_gcc_javaThread.cpp := stringop-overflow, \
+    DISABLED_WARNINGS_gcc_jvmtiRawMonitor.cpp := stringop-overflow, \
     DISABLED_WARNINGS_clang := $(DISABLED_WARNINGS_clang), \
     DISABLED_WARNINGS_clang_arguments.cpp := missing-field-initializers, \
     DISABLED_WARNINGS_clang_codeBuffer.cpp := tautological-undefined-compare, \
