diff --git a/src/hotspot/share/utilities/forbiddenFunctions.hpp b/src/hotspot/share/utilities/forbiddenFunctions.hpp
index 67e45037f62..60ca58014ea 100644
--- a/src/hotspot/share/utilities/forbiddenFunctions.hpp
+++ b/src/hotspot/share/utilities/forbiddenFunctions.hpp
@@ -25,6 +25,12 @@
 #ifndef SHARE_UTILITIES_FORBIDDENFUNCTIONS_HPP
 #define SHARE_UTILITIES_FORBIDDENFUNCTIONS_HPP
 
+#ifdef _WINDOWS
+#include "forbiddenFunctions_windows.hpp"
+#else
+#include "forbiddenFunctions_posix.hpp"
+#endif
+
 #include "utilities/compilerWarnings.hpp"
 #include "utilities/macros.hpp"
 
@@ -32,11 +38,7 @@
 #include <stddef.h> // for size_t
 #include <stdlib.h> // clang workaround for exit, _exit, _Exit - see FORBID macro.
 
-#ifdef _WINDOWS
-#include "forbiddenFunctions_windows.hpp"
-#else
-#include "forbiddenFunctions_posix.hpp"
-#endif
+
 
 // Forbid the use of various C library functions.  Some of these have os::
 // replacements that should be used instead.  Others are considered obsolete
