diff --git a/test/jdk/java/lang/Thread/virtual/ThreadPollOnYield.java b/test/jdk/java/lang/Thread/virtual/ThreadPollOnYield.java
index b8ab61556ea..2c4bf049971 100644
--- a/test/jdk/java/lang/Thread/virtual/ThreadPollOnYield.java
+++ b/test/jdk/java/lang/Thread/virtual/ThreadPollOnYield.java
@@ -40,14 +40,17 @@
  */
 
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.CountDownLatch;
 
 import jdk.test.lib.thread.VThreadPinner;
 import org.junit.jupiter.api.Test;
 import static org.junit.jupiter.api.Assertions.*;
 
 class ThreadPollOnYield {
+    public static CountDownLatch latch = new CountDownLatch(1);
     static void foo(AtomicBoolean done) {
         while (!done.get()) {
+            latch.countDown();
             Thread.yield();
         }
     }
@@ -58,7 +61,7 @@ void testThreadYieldPolls() throws Exception {
         var vthread = Thread.ofVirtual().start(() -> {
             VThreadPinner.runPinned(() -> foo(done));
         });
-        Thread.sleep(5000);
+        latch.await();
         done.set(true);
         vthread.join();
 
